src/gui/toolbarMenubar/ToolWindowOpacitySlider.cpp

```diff
@@ -0,0 +1,130 @@
+#include "ToolWindowOpacitySlider.h"
+
+#include "control/Control.h"
+#include "gui/MainWindow.h"
+
+#include "i18n.h"
+
+const double MIN_OPACITY = 0.2;
+const double MAX_OPACITY = 1.0;
+const double OPACITY_STEP = 0.1;
+
+class ToolWindowOpacitySlider::Impl {
+public:
+    Impl(bool horizontal, IconNameHelper iconNameHelper, Control* ctrl);
+
+    GtkToolItem* createItem(bool horizontal);
+    GtkToolItem* createTmpItem(bool horizontal);
+    GtkToolItem* newItem();
+    GtkWidget* getNewToolIcon() const;
+
+    void setEnabled(bool enabled);
+
+    static void sliderChanged(GtkRange* range, Impl* self);
+
+private:
+    bool horizontal_;
+    IconNameHelper iconNameHelper_;
+
+    GtkWidget* slider_ = nullptr;
+    GtkToolItem* toolItem_ = nullptr;
+    Control* control_ = nullptr;
+};
+
+// TODO: Memory management. Who is responsible for freeing slider_, toolItem_, ...?
+ToolWindowOpacitySlider::~ToolWindowOpacitySlider() = default;
+
+ToolWindowOpacitySlider::ToolWindowOpacitySlider(ActionHandler* handler, std::string id, ActionType type,
+                                                 Control* control, IconNameHelper iconNameHelper):
```

Comment:
In `ToolMenuHandler::initToolItems`, `control->getWindow()` returns `nullptr`. As such, we can't pass the `MainWindow*` to the `ToolWindowOpacitySlider` directly. I don't like giving it the entire `control` object, though...